checkstyle/src/main.rs

```diff
@@ -0,0 +1,124 @@
+use regex::Regex;
+use std::fs::{self};
+
+fn get_file(filename: &str) -> String {
+    fs::read_to_string(filename).expect("Failed to read file")
+}
+
+fn split_into_lines(s: &str) -> Vec<&str> {
+    s.lines().collect()
+}
+
+fn check_matching_regex_line(file: &str, re: Regex) -> Option<bool> {
+    let file_lines = split_into_lines(file);
+    for file_line in file_lines {
+        if re.is_match(file_line) {
+            return Some(true);
+        }
+    }
+    Some(false)
+}
+
+fn check_matching(file: &str, regex: &str) -> Option<bool> {
+    let lines: Vec<&str> = split_into_lines(regex);
+    for regex_line in lines {
+        let re = Regex::new(regex_line).ok()?;
+        if check_matching_regex_line(file, re) == Some(false) {
+            return Some(false);
+        }
+    }
+    Some(true)
+}
```

Comment:
I may be misreading this, but doesn't this just check that each regex line matches _a_ line in the file, but not necessarily in the right order?